src/custom/utils/signatures.ts

```diff
@@ -0,0 +1,72 @@
+import { domain as domainGp, signOrder as signOrderGp, Order } from '@gnosis.pm/gp-v2-contracts'
+import { ChainId } from '@uniswap/sdk'
+
+import { GP_SETTLEMENT_CONTRACT_ADDRESS } from 'constants/index'
+import { TypedDataDomain, Signer } from 'ethers'
+
+export { OrderKind } from '@gnosis.pm/gp-v2-contracts'
+export type UnsignedOrder = Order
+
+export interface SignOrderParams {
+  chainId: ChainId
+  signer: Signer
+  order: UnsignedOrder
+}
+
+// TODO: We cannot make use of the NPM exported enum, see https://babeljs.io/docs/en/babel-plugin-transform-typescript#caveats
+//      use workaround?
+// /**
+//  * The signing scheme used to sign the order.
+//  */
+// export declare const enum SigningScheme {
+//   /**
+//    * The EIP-712 typed data signing scheme. This is the preferred scheme as it
+//    * provides more infomation to wallets performing the signature on the data
+//    * being signed.
+//    *
+//    * <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator>
+//    */
+//   TYPED_DATA = 0,
+//   /**
+//    * The generic message signing scheme.
+//    */
+//   MESSAGE = 1
+// }
+
+// /**
+//  * Order kind.
+//  */
+// export declare const enum OrderKind {
+//   /**
+//    * A sell order.
+//    */
+//   SELL = 0,
+//   /**
+//    * A buy order.
+//    */
+//   BUY = 1
+// }
+
+// TODO: For now, instead of using enums (see todo above)
+const TYPED_DATA_SIGNING_SCHEME = 0
+export const ORDER_KIND_SELL = 0
+export const ORDER_KIND_BUY = 0
```

Comment:
Should this not be a strict [according to api](https://github.com/gnosis/gp-v2-services/blob/1ae1f46c54b2262f63b4178a9e935e56eca88a83/orderbook/openapi.yml#L149-L150)
Or is this for signing only,not for posting as JSON?

Reply:
Good point, but I think the issue is in the API definition. We are using the object as defined in the contracts, that is what you sign, so although it makes a worse API (less semantic), It goes in the direction of sending the API the raw unsigned data plus the signature.

@josojo I think this fields are slightly different. Should we copy it in the API (like we did with the epoch for the validTo?)


https://github.com/gnosis/gp-v2-contracts/blob/main/src/ts/order.ts#L92